pandas/core/ops.py

```diff
@@ -257,30 +256,87 @@ def add_flex_arithmetic_methods(cls, flex_arith_method,
                 exclude=exclude)
 
 
-class _TimeOp(object):
+class _Op(object):
+
     """
-    Wrapper around Series datetime/time/timedelta arithmetic operations.
-    Generally, you should use classmethod ``maybe_convert_for_time_op`` as an
-    entry point.
+    Wrapper around Series arithmetic operations.
+    Generally, you should use classmethod ``_Op.get_op`` as an entry point.
+
+    This validates and coerces lhs and rhs depending on its dtype and
+    based on op. See _TimeOp also.
+
+    Parameters
+    ----------
+    left : Series
+        lhs of op
+    right : object
+        rhs of op
+    name : str
+        name of op
+    na_op : callable
+        a function which wraps op
     """
-    fill_value = iNaT
+
+    fill_value = np.nan
     wrap_results = staticmethod(lambda x: x)
     dtype = None
 
     def __init__(self, left, right, name, na_op):
+        self.left = left
+        self.right = right
+
+        self.name = name
+        self.na_op = na_op
+
+        self.lvalues = left
+        self.rvalues = right
+
+    @classmethod
+    def get_op(cls, left, right, name, na_op):
+        """
+        Get op dispatcher, returns _Op or _TimeOp.
+
+        If ``left`` and ``right`` are appropriate for datetime arithmetic with
+        operation ``name``, processes them and returns a ``_TimeOp`` object
+        that stores all the required values.  Otherwise, it will generate
+        either a ``_Op``, indicating that the operation is performed via
+        normal numpy path.
+        """
+        is_timedelta_lhs = is_timedelta64_dtype(left)
+        is_datetime_lhs = (is_datetime64_dtype(left) or
+                           is_datetime64tz_dtype(left))
 
-        # need to make sure that we are aligning the data
         if isinstance(left, ABCSeries) and isinstance(right, ABCSeries):
-            left, right = left.align(right, copy=False)
+            # avoid repaated alignment
+            if not left.index.equals(right.index):
+                left, right = left.align(right, copy=False)
+
+                index, lidx, ridx = left.index.join(right.index, how='outer',
+                                                    return_indexers=True)
+                # convert DatetimeIndex to have UTC
```

Comment:
Sorry, still not fully following. Doesn't the alignment already take care of that?

Using your example from above:

```
In [38]: left = pd.Series([1,2], index=idx1)

In [39]: right = pd.Series([1,2], index=idx2)

In [40]: left, right = left.align(right, copy=False)

In [41]: left.index
Out[41]: DatetimeIndex(['2011-01-01', '2011-01-02', '2011-01-03'], dtype='dateti
me64[ns, UTC]', freq=None)

In [42]: right.index
Out[42]: DatetimeIndex(['2011-01-01', '2011-01-02', '2011-01-03'], dtype='dateti
me64[ns, UTC]', freq=None)
```

Also, the `lidx` and `ridx` does not seem to get used here